test/performance/ThroughputPublisher.cpp

```diff
@@ -322,23 +373,42 @@ void ThroughputPublisher::run(uint32_t test_time, uint32_t recovery_time_ms, int
         for (auto dit = sit->second.begin(); dit != sit->second.end(); ++dit)
         {
             std::this_thread::sleep_for(std::chrono::milliseconds(100));
-            //cout << "Starting test with demand: " << *dit << endl;
             command.m_command = READY_TO_START;
             command.m_size = sit->first;
             command.m_demand = *dit;
-            //cout << "SEND COMMAND "<< command.m_command << endl;
+
+            // Check history resources depending on the history kind and demand
+            if (pubAttr.topic.historyQos.kind == KEEP_LAST_HISTORY_QOS)
+            {
+                // Ensure that the history depth is at least the demand
+                if (pubAttr.topic.historyQos.depth < 0 ||
+                    static_cast<uint32_t>(pubAttr.topic.historyQos.depth) < command.m_demand)
```

Comment:
Not sure about this. What will happen in the case that `historyQos.depth` (not sure if this is possible, as we could be checking for this in the code) is negative when we cast to to `uint32_t`? I would re-write the above if as:

`if (pubAttr.topic.historyQos.depth < static_cast<int32_t>(command.m_demand)`

I believe only this condition is enough. The same applies for the `KEEP_ALL` case below.

Reply:
OK, as discussed, that would never happen as the first condition will be checked first.